### ISSUE TITLE
Consider removing warnings and failsafe checks

### ISSUE THREAD

**Issue by AaronFriel** [pulumi/pulumi-docker#846]
In PR #843, we add a number of checks to ensure that the image built is the one we expect, and that the repo digest returned from running the push is what we expect. As a follow up, we could consider a couple changes after getting broader feedback:
- [ ] Remove Pulumi build ID label and polling the image store around `ImageBuild`.
      
  If no users report seeing an error that we failed to find the correct image in the store, we can rule out a race condition between ImageBuild and ImagePush, and remove the polling the image store.
- [ ] Remove `getRepoDigest` in favor of the `ImagePush` result.

  If no users report that the expected repo digest, obtained from `ImagePush`, differs with the actual repo digest, returned from `ImageInspectWithRaw`, then we can treat a missing digest from `ImagePush` as an error.

**Comment by mjeffryes** [comment]
Removed the Pulumi build ID label in #1225

**Comment by BackSlasher** [comment]
## Reporting the divergence case mentioned in this issue

This issue's second checkbox said:

> If no users report that the expected repo digest, obtained from `ImagePush`, differs with the actual repo digest, returned from `ImageInspectWithRaw`, then we can treat a missing digest from `ImagePush` as an error.

Filing the divergence report — concrete repro against podman, evidence below.

### Setup

- Docker-compatible daemon: **podman 5.8.2 system service** (also reproduced against a patched podman 6.0.0-dev build to surface `aux` on the stream).
- Registry: Google Artifact Registry, `me-west1-docker.pkg.dev`.
- Resource: `docker.Image` from `@pulumi/docker@5.0.0`, build context `FROM alpine:3.20`, registry auth via gcloud oauth2 access token.

### Divergence

| Source | Digest |
|---|---|
| `aux` from `ImagePush` stream (parsed into `expectedRepoDigest`) | `sha256:9566f2f0e748749e795b694a6b4c27b58b180d7639c596e03395b3da475c2d83` |
| `getRepoDigest` from `ImageInspect` after push | `sha256:91f2477c71550957169896e6471a1bf8e2683f40bf844cdfd33a59ccc1b4eed8` |
| `gcloud artifacts docker images describe ...:probe-1` (registry truth) | `sha256:9566f2f0...` (matches `aux`) |

Provider warning fired exactly as PR #843 designed it to:

```
warning: Expected repo digest "me-west1-docker.pkg.dev/.../test-test@sha256:9566f2f0...",
                         found "me-west1-docker.pkg.dev/.../test-test@sha256:91f2477c..."
```

The `aux` value is the correct registry-side digest. The local-inspect value is wrong — the registry has never seen it:

```
$ podman pull me-west1-docker.pkg.dev/.../test-test@sha256:91f2477c...
Error: ... reading manifest sha256:91f2477c... in me-west1-docker.pkg.dev/.../test-test:
manifest unknown: Requested entity was not found.

$ podman pull me-west1-docker.pkg.dev/.../test-test@sha256:9566f2f0...
# works
```

Direct registry HEAD on `manifests/sha256:91f2477c...` returns 404 under every Accept type (Docker v2, Docker v2 list, OCI manifest, OCI index). It is not an index / manifest-list situation — the digest simply doesn't exist at the registry.

### Root cause and upstream context

Podman re-serializes the manifest during push (key order / whitespace / canonicalization), so the on-the-wire bytes hash differently from the locally-stored manifest bytes. The local image store keeps the pre-serialization digest as the image's `Digest` and never reconciles it with what was actually pushed, even though `pkg/domain/infra/abi/images.go` *knows* the right value (`manifest.Digest(pushedManifestBytes)`) and returns it as `ImagePushReport.ManifestDigest` — the same value that ends up in `aux`.

This is tracked in podman as [containers/podman#14779](https://github.com/containers/podman/issues/14779) (open since 2022) and at the buildah layer as [containers/buildah#3866](https://github.com/containers/buildah/issues/3866). The podman maintainers' position (mtrmac in [#14779](https://github.com/containers/podman/issues/14779#issuecomment-1549159527)) is that `push && inspect` is fundamentally an under-specified pattern because an image can have multiple legitimate digests across registries and compression representations, and that consumers should use `podman push --digestfile <path>` (or the equivalent push-result digest in the API stream) instead of inspecting the local image. That position lines up directly with this issue's second checkbox.

### Downstream impact

The exported `repoDigest` output is what users feed into immutability-pinned consumers — `gcp.cloudrunv2.Service.containers[].image`, `aws.lambda.Function.imageUri`, Sigstore attestations, etc. With the local-inspect value, those consumers get the wrong digest and 404 against the registry on next rollout. That's the same use case that motivates having `repoDigest` as an output at all.

One caveat worth flagging: vanilla podman doesn't currently emit `aux` on the push stream at all — the "Expected X, found Y" warning above only appears once you patch podman to emit it. We're planning to send a podman PR for that separately (it's a small change in `pkg/api/handlers/compat/images_push.go` — `pushReport.ManifestDigest` is already known there, it just isn't wrapped into the `aux` JSON object the Docker daemon emits). On current podman, users silently get a wrong `repoDigest` output with only the "Push completed without reporting a digest" warning, which makes this divergence invisible — they see the cosmetic warning, not the "Expected X, found Y" one that would actually flag the problem.

### Proposed action

Pick up the unchecked checkbox: prefer `expectedRepoDigest` (the `aux` value) over `getRepoDigest`'s `ImageInspect` result when present; fall back to inspect when `aux` is absent. Treat persistent absence of `aux` as the existing warning, not an error, so daemons that don't emit it yet (podman today) still work.

This also aligns with the upstream podman maintainers' position that the push-result digest is the authoritative one.

Happy to send a PR if you'd like — let me know if you'd prefer a fresh issue rather than reopening this one.
